config/config.example.yml

```diff
@@ -77,6 +77,11 @@ submission:
     # NOTE: after how many time (milliseconds) submission is saved automatically
     # eg. timer: 5 * (1000 * 60); // 5 minutes
     timer: 0
+  typeBind:
+    # NOTE: which field to use when matching to type-bind configuration,
+    # eg. dc.type, local.publicationType
+    # default: dc.type
+    field: dc.type
```

Comment:
@tdonohue the cleanest way i found to do this is to have the config reading in the Form Builder service - this is provided by so many components though, so adding ConfigurationDataService as another dependency means updating a lot of tests and components to provide this, too (or a mock version in the case of tests).
I got about half way through this and wondered if it's worth it... should I instead allow Form Builder to *not* be provided ConfigDataService in which case we'll nicely handle that fact and either return a default "dc.type" type field, or null, or something - but not a fatal error. Or is it better to update all these components? Many of them are not related to submission but in the future may still like to read backend config values or even use type binding.

Reply:
hm, actually i might have a better way if i play with optional data service, and keep any setting / init out of the constructor so only components that care can provide it

Reply:
@tdonohue I've add this in, there is now a getTypeField() method in the FormBuilderService. When it is called, and the (optional) config data service is present, the local typeField will subscribe to the configured value from the REST API, and return it. Otherwise it'll return a default of dc_type.
Making the config data service optional was my approach to avoid refactoring many related components and tests that use form builder outside of submission. If it is preferred to wire configuration data service into all the providers that include FormBuilderService instead, I can do this, it's just a bigger change that I wanted to avoid. Happy to hear feedback!

Reply:
@tdonohue I've added tests for isObjectEmpty, should be good now

Reply:
Thanks for the update @kshepherd .  I'll give this another review/test once the backend PR is fully updated.  Overall, your approach to resolving this config issue seems fine to me.  I don't have objections.